backend/src/main/controller/UserController.ts

```diff
@@ -0,0 +1,104 @@
+import { Request, Response, NextFunction, response } from "express";
+import { getCustomRepository, getRepository } from "typeorm";
+import { User } from "../entity/User";
+import { UserRepository } from "../repository/UserRepository";
+import { IdValidation } from "../validation/IdValidation";
+import { UserDTO, UserValidation } from "../validation/UserValidation";
+import { UserUpdateValidation, UserUpdateDTO} from "../validation/UserUpdateValidation";
+
+export class UserController {
+    /**
+     * Access user profile data
+     * * validate query first
+     * * apply them into query builder
+     */
+
+    // return all user
+    listAll = async (req: Request, res: Response, next: NextFunction) => {
+        try {
+            // get users from database
+            const userRepository = getCustomRepository(UserRepository);
+            const users = await userRepository.find({
+                select: ["id", "username", "firstName", "lastName"]
+            });
+    
+            // send the users object
+            res.send(users);
+        } catch (e) {
+            return next(e);
+        }
+    };
+
+    // return one user
+    getOneById = async (req: Request, res: Response, next: NextFunction) => {
+        try {
+            // get the id from the url
+            const id = await IdValidation.schema.validateAsync(req.params.id, {
+                abortEarly: false,
+            });
+            const userRepository = getCustomRepository(UserRepository);
+            const user = await userRepository.findOneOrFail(id, {
```

Comment:
Oh wait for this one, do you want to try to add `select: false`?
https://github.com/ubclaunchpad/foodies/blob/6eb159a5c29916d0b99331ad452432d13b1c7c03/backend/src/main/entity/User.ts#L68-L72

Then you can just do `userRepository.findOneOrFail(id)` without the select.